tests/api_connexion/endpoints/test_dataset_endpoint.py

```diff
@@ -80,7 +80,9 @@ def test_should_respond_200(self, session):
         assert session.query(DatasetModel).count() == 1
 
         with assert_queries_count(5):
-            response = self.client.get("/api/v1/datasets/1", environ_overrides={'REMOTE_USER': "test"})
+            response = self.client.get(
+                "/api/v1/datasets/s3%3A%2F%2Fbucket%2Fkey", environ_overrides={'REMOTE_USER': "test"}
```

Comment:
nit: Might be better to have unencoded uri here and encode it, like this: https://github.com/apache/airflow/pull/25774/files#diff-b9ec718a601d8a76e6a2cd05b826593cc761fbef74c2bbd1c8f2ba0e5e8eb603R133